lib/bin/run_test.js

```diff
@@ -51,9 +51,13 @@ function invokeGeneratorOrPromiseFunction(fun) {
   }
 
   if (result && result.next) {  // This looks like a generator
-    return co(result);
+    return when.promise(function(resolve) {
+      // fixme: the promise returned by co seems to be
+      //        incompatible with the ones that when uses.
+      co(result).then(resolve);
```

Comment:
I think a better way of writing this is `return when(co(result));`. I'm not 100% sure that will work, but I think this way of doing it won't. You will at least have to pass in a `reject` parameter to the `then` call.